src/python/grpcio/grpc/_server.py

```diff
@@ -828,7 +828,9 @@ def stop(self, grace):
         return _stop(self._state, grace)
 
     def __del__(self):
-        _stop(self._state, None)
+        if hasattr(self, '_state'):
```

Comment:
This change is required to pass the unit test, otherwise there is no way to destroy C-Core server object. It suppose to be a strictly better modification. The worse case of this change is that the `del` call didn't trigger garbage collect.

Another alternative way is to pause this PR, and clean-up the server-side object destroy procedure first?

Reply:
This is not the same content as https://github.com/grpc/grpc/pull/17259, which still needs further review/investigation. Please don't embed one active PR into another without calling attention to this fact somewhere (e.g., "this depends on a bug fix in #...").

Another option would be to adjust the unit tests here to take into account current (albeit suboptimal) behavior, which we believe to still be possible even with the call to `del` here anyways. How many tests rely on this?

Reply:
5 tests maybe affected by this change, all the tests involving a gRPC server. I don't think testing based on a faulty behavior is a good idea. It is hackable to make it pass. We can count how many server is created and use a counter to label as removed.

The `del` need to be called during the deconstruction process to trigger the `__dealloc__` in Cython. Calling the `del _server._state` inside the unit test will be a noop. Also, the added condition is for checking the existence of `_state`, without it, CPython will complain:
```
Exception AttributeError: "'_Server' object has no attribute '_state'" in <bound method _ChannelServerPair.__del__ of <__main__._ChannelServerPair object at 0x109b7e090>> ignored
Exception AttributeError: "'_Server' object has no attribute '_state'" in <bound method _Server.__del__ of <grpc._server._Server object at 0x109b7e110>> ignored
Exception AttributeError: "'_Server' object has no attribute '_state'" in <bound method _Server.__del__ of <grpc._server._Server object at 0x109b7ea50>> ignored
```
I didn't add this check in #17259, because Google version of Python doesn't complain.

Do you think adding a comment can be a valid option here?
```
# TODO(lidiz): Depends on issue #17258 which is not solved yet
```